src/core/inventory.ts

```diff
@@ -0,0 +1,371 @@
+import { Address } from 'viem'
+import { chains, getAssetWeight } from '../config/chains'
+import { getPublicClient } from '../helpers/getClients'
+
+// TODO: Maybe move to an artifacts directory?
+// Minimal ERC20 ABI - only what we need for balance checking (and decimals)
+const erc20ABI = [
+  {
+    constant: true,
+    inputs: [{ name: '_owner', type: 'address' }],
+    name: 'balanceOf',
+    outputs: [{ name: 'balance', type: 'uint256' }],
+    type: 'function',
+  },
+  {
+    constant: true,
+    inputs: [],
+    name: 'decimals',
+    outputs: [{ name: '', type: 'uint8' }],
+    type: 'function',
+  },
+] as const
+
+// Stub prices for assets (in USD)
+const assetPrices: Record<string, number> = {
```

Comment:
how will this work properly in prod? would we cache prices and have a job to update them?

Reply:
That would be the most ideal form! I just wanted to get the naive implementation in here, see response to earlier comment: https://github.com/rhinestonewtf/rhinestone-relayer/pull/44#discussion_r2062702732